raiden-cli/src/routes/channels.ts

```diff
@@ -52,7 +64,91 @@ async function openChannel(this: Cli, request: Request, response: Response, next
   } catch (error) {
     if (isInvalidParameterError(error)) {
       response.status(400).send(error.message);
-    } else if (error.code === 'INSUFFICIENT_FUNDS') {
+    } else if (isInvalidParameterError(error)) {
+      response.status(402).send(error.message);
+    } else if (isConflictError(error)) {
+      response.status(409).send(error.message);
+    } else {
+      next(error);
+    }
+  }
+}
+
+const allowedUpdateKeys = new Set(['state', 'total_deposit', 'total_withdraw']);
+function validateChannelUpdateBody(request: Request, response: Response, next: NextFunction) {
+  const intersec = Object.keys(request.body).filter((k) => allowedUpdateKeys.has(k));
+  if (intersec.length < 1)
+    return response
+      .status(400)
+      .send('one of "state" | "total_deposit" | "total_withdraw" operations required');
+  else if (intersec.length > 1)
+    return response
+      .status(409)
+      .send('more than one of "state" | "total_deposit" | "total_withdraw" requested');
+  if (
+    request.body.state &&
+    ![ApiChannelState.closed, ApiChannelState.settled].includes(request.body.state)
+  )
+    return response
+      .status(400)
+      .send('invalid "state" requested: must be one of "closed" | "settled"');
+  return next();
+}
+
+async function updateChannel(this: Cli, request: Request, response: Response, next: NextFunction) {
```

Comment:
This function mixes different kind of abstractions. The higher level purpose of this function is to get the channel at its current state, choose the correct kind of update and return the updated channel. Instead you have to deal with all the details for each kind of update while reading it. I propose to export all the kind of updates (change state, increase deposit, withdraw) into their own functions while each of them will return the updated channel or throw some errors. This function here will feed them with the channel and the query parameter to update  (e.g. the target state or the deposit amount). At the end this function is responsible to evaluate the thrown error and respond to the user in general.

Sorry that I always bug you with splitting your functions. But I think there are many simple rules to achieve qualitative good code. And I barely see good reasons why not to try following them. I mean there are also way more strict rules. Robert C. Martin for examples thinks that a function should be so focused that even error handling (i.e. try-catch) is a single responsibility. I wouldn't go so far. I'm just asking for some basics to keep the code better readable, since I think this is key to good software.